tools/querytee/proxy_backend.go

```diff
@@ -85,6 +117,36 @@ func (b *ProxyBackend) Preferred() bool {
 	return b.preferred
 }
 
+func (b *ProxyBackend) RequestProportion() float64 {
+	return b.requestProportion
+}
+
+func (b *ProxyBackend) SetRequestProportion(proportion float64) {
+	b.requestProportion = proportion
+}
+
+func (b *ProxyBackend) HasConfiguredProportion() bool {
+	return b.cfg.RequestProportion != nil
+}
+
+func (b *ProxyBackend) MinTimeThreshold() time.Duration {
+	return b.minTimeThreshold
+}
+
+func (b *ProxyBackend) ShouldHandleQuery(minQueryTime time.Time) bool {
+	// If the time received is zero we don't know the time so we serve the request
+	if minQueryTime.IsZero() {
+		return true
+	}
+	// If threshold is 0s, backend serves all queries
+	if b.minTimeThreshold == 0 {
+		return true
+	}
+	// Backend serves queries where min_query_time >= (now - threshold)
+	cutOffTs := time.Now().Add(-b.minTimeThreshold)
+	return !minQueryTime.Before(cutOffTs)
```

Comment:
I believe this condition is inverted, should be `minQueryTime.Before(cutOffTs)`. The tests are also inverted to play along, e.g. `"instant query with old time - only preferred backend selected"`. We actually want all backends for "old" data queries.